src/components/chipsInput/index.js

```diff
@@ -96,27 +96,32 @@ class ChipsInput extends Component {
       value: props.value,
       tags: _.cloneDeep(props.tags) || [],
       tagIndexToRemove: undefined,
-      initialTags: props.tags
+      initialTags: props.tags,
+      inputRef: undefined
     };
   }
 
   componentDidMount() {
     if (Constants.isAndroid) {
-      const textInputHandle = ReactNative.findNodeHandle(this.input);
+      const textInputHandle = ReactNative.findNodeHandle(this.state.inputRef);
       if (textInputHandle && NativeModules.TextInputDelKeyHandler) {
         NativeModules.TextInputDelKeyHandler.register(textInputHandle);
       }
     }
   }
 
   static getDerivedStateFromProps(nextProps, prevState) {
+    const nextState = {}
     if (nextProps.tags !== prevState.initialTags) {
-      return {
-        initialTags: nextProps.tags,
-        tags: nextProps.tags
-      };
+      nextState.initialTags = nextProps.tags;
+      nextState.tags = nextProps.tags;
     }
-    return null;
+    if (nextProps.tags.length > prevState.initialTags.length) {
+      nextState.value = ''
+      prevState.inputRef && prevState.inputRef.clear()
```

Comment:
This is not the place to clear the input. Use `componentDidUpdate`.

Reply:
so I should clear the state's value in the derivedProps and clear the input in didMount?

Reply:
@Inbal-Tish

Reply:
In `ComponentDidUpdate` when meeting your condition.

Reply:
So I'll move only the ref clean to `DidUpdate`. okay?

Reply:
Sure. But I still don't understand what you're trying to solve and why clear the value will help you do that? Can you please add it to the description. Screenshots/videos will help.